packages/core/src/agent/index.ts

```diff
@@ -417,6 +417,7 @@ export class Agent<
   sanitizeResponseMessages(messages: Array<CoreMessage>): Array<CoreMessage> {
     let toolResultIds: Array<string> = [];
     let toolCallIds: Array<string> = [];
+    let dupeToolCalls = new Set<string>();
```

Comment:
You should move to Set instead of array

Reply:
no when i checked it it said array :D

Reply:
or maybe i was blind

Reply:
no it was like that and then I pushed up another commit haha